docs/advanced-guides/oauth-client.md

```diff
@@ -0,0 +1,293 @@
+- the `ES256` cryptographic system must be supported by all clients and auth servers
+- JWT header fields include:
+    - `alg`: `ES256`
+    - `kid`: string indicating which of the declared keys (in JWKS) is being used
+- JWT body fields include:
+    - `iss`: the `client_id`
+    - `sub`: the `client_id`
+    - `aud`: the authorization server issuer URL (origin)
+    - `jti`: randomly generated token string
+    - `iat`: current UNIX time (integer seconds)
+
+### Token Management
+
+Long-lived clients will need to manage access token lifetimes and periodic refresh token requests. This is functionality that is sometimes implemented in generic OAuth libraries. Care must be taken to ensure that concurrent resource requests don't result in concurrent token refresh requests, which could result in errors and loss of the overall auth session (requiring re-authorization).
+
+Access and refresh tokens should never be copied or shared across end devices. They should not be stored in session cookies.
+
+### Hardened HTTP Client (SSRF)
+
+Clients need to make a several HTTP network requests using URLs provided by unknown parties. These raise a number of security concerns, including network requests to local or private IP addresses (SSRF), and trivial Denial of Service attacks (large response bodies, slow responses, etc).
+
+A good way to mitigate these issues is to use or implement a hardened HTTP client. It should set appropriate timeouts and resource limits, validate URLs, and check that resolve domain names don't point to protected or local IP addresses.
+
+## Authorization Flow
+
+### Account or Server Identifier
+
+Clients can start an auth flow in one of two ways:
+
+- starting with an atproto account identifier: handle or DID
+- starting with a server URL or hostname (PDS or entryway)
+
+One use case for starting with a server URL instead of an account identifier is when the user does not remember their full account handle or only knows their account email. Another is for authentication when a user’s handle is broken. The user still needs to know their hosting provider in these situations.
+
+When starting with an account identifier, the client must resolve the atproto identity to a DID document. If starting with a handle, it is critical (mandatory) to bidirectionally verify the handle by checking that the DID document claims the handle (see atproto Handle specification). All handle resolution techniques and all atproto-blessed DID methods must be supported to ensure interoperability with all accounts.
+
+In some client environments, it may be difficult to resolve all identity types. For example, handle resolution may involve DNS TXT queries, which are not directly supported from browser apps. Client implementations might use alternative techniques (such as DNS-over-HTTP) or could make use of a supporting web service to resolve identities.
```

Comment:
```suggestion
In some client environments, it may be difficult to resolve all identity types. For example, handle resolution may involve DNS TXT queries, which are not directly supported from browser apps. Client implementations might use alternative techniques (such as DNS-over-HTTPS) or could make use of a supporting web service to resolve identities.
```